src/metabase/models/setting.clj

```diff
@@ -1208,7 +1229,7 @@
         setting-setter-fn-symbol (-> (symbol (str (name setting-symbol) \!))
                                      (with-meta (meta setting-symbol)))
         setting-definition-symbol (gensym "setting-")]
-    `(let [setting-options#          (merge ~options ~setting-metadata)
+    `(let [setting-options#          (merge (#'inherit-base ~options) ~setting-metadata)
```

Comment:
```suggestion
    `(let [setting-options#          (merge (inherit-base ~options) ~setting-metadata)
```

IMO, I'd rather make `inherit-base` public than using the var-quote like this.

Reply:
I'm slightly torn, but will make the change. I don't feel that this belongs in the public API, but I also don't like exposing a "var" call in generated code, both aesthetically amd as it has performance overhead, although its mute for a form that only executes during compilation.

Reply:
How about making `merge-base` a local variable?

Reply:
You know [how I feel about long functions](https://dev.to/kanani_nirav/the-five-lines-of-code-principle-why-less-is-more-in-programming-31j6) - you see I'm a recovering old school Ruby programmer. Having more code emitted in macro expansion also makes me uncomfortable about code bloat, but this is probably trauma from C++ and Rust where it can have a big impact on compile time and binary sizes. 

You may still have won me over that it's better than emitting code with `#'` though 😄 

I've made peace with just having it public though. Since merge is no longer recursive, exposing it gives users a way to do their own transitive extensions.